### PR TITLE
Fix doc typo

### PR DESCRIPTION
**Pull request by Atcold**
Fix bad formatting.

### CODE REVIEW

**Comment by Atcold** [comment]
Hold on before merging. I'm finding other typos and stuff.

**Comment by lvdmaaten** [comment]
Ready to be merged?

**Comment by Atcold** [comment]
@lvdmaaten: haha, I'm still fixing stuff. But I could open another PR for that.
One question. Why don't you use `optim`'s `Logger`, which allows you to plot the charts with `gnuplot`?

**Comment by lvdmaaten** [comment]
The torchnet `tnt.Log` is way more general: it lets you do remote logging, gives you complete control over logging format, lets you log in multiple formats at a time, etc. If you want to plot some of your logged values, that can be trivially done.

The `optim.Logger` doesn't make any sense to me: what business has a logger (or a confusion matrix, for that matter) in an optimization package? And why does logging code depend on plotting code? Even worse: who adds `gnuplot` dependencies in an optimization package? The goal of `torchnet` is to encourage people to write modular code, which means not writing this kind of insanity.

**Comment by Atcold** [comment]
"that can be trivially done" -> I guess we'll have to add this to the doc.
Btw, you can merge this, and I can open a second doc-fix PR, so that people can start using the new doc. What do you think?

File: README.md [hunk 1]
```diff
@@ -1463,24 +1463,25 @@ events, and will print out only status records.
...
+tnt = require 'torchnet'
 
 -- require the viewers we want
-local logtext = require 'torchnet.log.view.text'
-local logstatus = require 'torchnet.log.view.status'
+logtext = require 'torchnet.log.view.text'
+logstatus = require 'torchnet.log.view.status'
 
-local log = tnt.Log{
+log = tnt.Log{
```
Comment (lvdmaaten) [inline README.md]: Why remove all the `local`s here. I would encourage anyone to *never* use global variables. Ever.

File: README.md [hunk 2]
```diff
@@ -955,38 +955,78 @@ An instance `engine` of a `tnt.Engine()` implements two main methods:
...
 
 ### tnt.SGDEngine
 
-The SGDEngine module implements the Stochastic Gradient Descent training
+The `SGDEngine` module implements the Stochastic Gradient Descent training
 procedure in `train`, including data sampling, forward prop, back prop, and
 parameter updates. It also operates as a coroutine allowing a user control
  (i.e. increment some sort of `tnt.Meter`) at events such as 'start',
 'start-epoch', 'forward', 'forward-criterion', 'backward', etc.
+The available hooks are the following:
```
Comment (lvdmaaten) [inline README.md]: Actually, the hooks that are available are `Engine`-specific. Something like a `SVMEngine` (if we ever write it) may have very different hooks.

File: README.md [hunk 3]
```diff
@@ -955,38 +955,78 @@ An instance `engine` of a `tnt.Engine()` implements two main methods:
...
+loss function (`nn.Criterion`), a dataset iterator (`tnt.DatasetIterator`), and a
 learning rate, at the minimum. The `test` function allows for simple evaluation
 of a model on a dataset.
 
 A `state` is maintained for external access to outputs and parameters of modules
-as well as sampled data.
+as well as sampled data. The content of the `state` table is the following, where
+the passed values come from the arguments of `engine:train()`:
+```lua
+state = {
```
Comment (lvdmaaten) [inline README.md]: Similarly, the exact implementation of `state` is `Engine`-specific.

File: log/init.lua [hunk 4]
```diff
@@ -14,43 +14,44 @@ local doc = require 'argcheck.doc'
...
 events, and will print out only status records.
 
 A typical use case would be the following:
 ```lua
+tnt = require 'torchnet'
 
 -- require the viewers we want
-local logtext = require 'torchnet.log.view.text'
-local logstatus = require 'torchnet.log.view.status'
+logtext = require 'torchnet.log.view.text'
```
Comment (lvdmaaten) [inline log/init.lua]: Idem. Better leave the `local`s in.

**Comment by lvdmaaten** [comment]
Added a few comments. Can you please address these before I merge this in?

File: log/init.lua (hunk 4 above)
Comment (Atcold) [inline log/init.lua]: It won't run in the REPL... These are examples people copy & paste into the terminal.

File: README.md (hunk 3 above)
Comment (Atcold) [inline README.md]: OK, I'll add both `Engine`s' `state`s.

File: README.md (hunk 2 above)
Comment (Atcold) [inline README.md]: Oh... But now they are the same, for the two `Engine`s we have, right?

File: README.md (hunk 1 above)
Comment (Atcold) [inline README.md]: With `local`, the code won't run when copied in the REPL. It's not a script.

**Comment by Atcold** [comment]
Usually people copy and paste code segments from the documentation into the REPL just to see how they execute. With the `local` the code will not run. Of course, `local`s are fundamental in scripts, but this is not the case here.
We've done the same with the whole Torch7 documentation. There are no `local`s whatsoever in the whole documentation.

**Comment by lvdmaaten** [comment]
You can just put a `do ... end` block around it if you want to run something with `local`s in the REPL? It's too bad the Torch7 documentation is encouraging folks to write bad code.

**Comment by Atcold** [comment]
If you use `local`s within a `do ... end`, then you will no longer have access to the variables.
It is a bad practice only when scripting. But it's the only way you can use the REPL is a sane way.
The REPL is thought to be an interactive interpreter. So that you can fool around and learn. It's not, by all means, a scripting proxy.

Therefore, in my opinion, in the documentation, examples should be runnable by beginners in the REPL.
If it does not run, then it makes learning harder.
It can be stated that *good practice* mandates the use of `local` when scripting.

Nevertheless, the repo is yours, and I will do as you decide.

**Comment by ebetica** [comment]
@facebook-github-bot import

**Comment by ebetica** [comment]
@facebook-github-bot import

**Comment by ebetica** [comment]
Sorry about spamming, testing automatic integration with Facebook's internal repo, which will let us push out updates to Torchnet much faster.

@facebook-github-bot shipit

**Comment by ebetica** [comment]
@Atcold looks like you have to complete the Facebook CLA for this to work.... https://code.facebook.com/cla

**Comment by facebook-github-bot-3** [comment]
I cannot do that @ebetica, because @Atcold does not have an active CLA on file.

**Pull request closed without merging**